src/libraries/Common/src/System/Security/Cryptography/CompositeMLDsa.cs

```diff
@@ -0,0 +1,1800 @@
+        ///   The data to sign.
+        /// </param>
+        /// <param name="destination">
+        ///   The buffer to receive the signature.
+        /// </param>
+        /// <param name="context">
+        ///   An optional context-specific value to limit the scope of the signature.
+        ///   The default value is an empty buffer.
+        /// </param>
+        /// <returns>
+        ///   The total number of bytes written to <paramref name="destination"/>.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        ///   The buffer in <paramref name="destination"/> is too small to hold the signature.
+        /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        ///   <paramref name="context"/> has a <see cref="ReadOnlySpan{T}.Length"/> in excess of
+        ///   255 bytes.
+        /// </exception>
+        /// <exception cref="ObjectDisposedException">
+        ///   This instance has been disposed.
+        /// </exception>
+        /// <exception cref="CryptographicException">
+        ///   <para>The instance represents only a public key.</para>
+        ///   <para>-or-</para>
+        ///   <para>An error occurred while signing the data.</para>
+        /// </exception>
+        /// <remarks>
+        ///   The signature will be at most <see cref="CompositeMLDsaAlgorithm.MaxSignatureSizeInBytes"/> in length.
+        /// </remarks>
+        public int SignData(ReadOnlySpan<byte> data, Span<byte> destination, ReadOnlySpan<byte> context = default)
+        {
+            if (destination.Length < Algorithm.MLDsaAlgorithm.SignatureSizeInBytes)
+            {
+                throw new ArgumentException(SR.Argument_DestinationTooShort, nameof(destination));
+            }
+
+            if (context.Length > MaxContextLength)
+            {
+                throw new ArgumentOutOfRangeException(
+                    nameof(context),
+                    context.Length,
+                    SR.Argument_SignatureContextTooLong255);
+            }
+
+            ThrowIfDisposed();
+
+            return SignDataCore(data, context, destination);
+        }
+
+        /// <summary>
+        ///   When overridden in a derived class, computes the signature of the specified data and context,
+        ///   writing it into the provided buffer.
+        /// </summary>
+        /// <param name="data">
+        ///   The data to sign.
+        /// </param>
+        /// <param name="context">
+        ///   The signature context.
+        /// </param>
+        /// <param name="destination">
+        ///   The buffer to receive the signature, whose length will be at least the ML-DSA component's signature size.
+        /// </param>
+        /// <exception cref="CryptographicException">
+        ///   An error occurred while signing the data.
+        /// </exception>
+        protected abstract int SignDataCore(ReadOnlySpan<byte> data, ReadOnlySpan<byte> context, Span<byte> destination);
```

Comment:
Because of the variable length from EC-DSA in the X.509 format, the abstract method probably wants to be `protected abstract bool TrySignDataCore(data, context, destination, out int bytesWritten)`.

Otherwise the Core method needs to throw an ArgumentException that destination was too small.

Reply:
Similarly, we will want a public TrySignData.  Thanks, IETF.